### PR TITLE
Rewrite build.sh to use CMake for PROJ 8.1.0

### LINKED ISSUE #105: Potentially build with cmake.
- [x] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:

Was wondering your thoughts on building this package with cmake instead of autoconf/tools. The cmake install provides cmake modules for configuration/use in downstream native projects which is super handy.

### PR DESCRIPTION
**Pull request by mwtoews**
#### Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

#### Summary of changes
* Closes #105 
* While PROJ currently support both Autotools and CMake builds, only the CMake build installs modules for CMake's `find_package()`, which can be handy for other projects.

### CODE REVIEW

**Comment by mwtoews** [comment]
All except `osx_arm64_` pass. The error isn't really obvious:
```
...
[ 76%] Building CXX object src/CMakeFiles/proj.dir/iso19111/operation/vectorofvaluesparams.cpp.o
[ 76%] Linking CXX shared library ../lib/libproj.dylib
ld: warning: -pie being ignored. It is only used when linking a main executable
[ 76%] Built target proj
make: *** [Makefile:160: all] Error 2
Traceback (most recent call last):
  File "/Users/runner/miniforge3/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/Users/runner/miniforge3/lib/python3.8/site-packages/conda_build/cli/main_build.py", line 481, in main
    execute(sys.argv[1:])
  File "/Users/runner/miniforge3/lib/python3.8/site-packages/conda_build/cli/main_build.py", line 470, in execute
    outputs = api.build(args.recipe, post=args.post, test_run_post=args.test_run_post,
  File "/Users/runner/miniforge3/lib/python3.8/site-packages/conda_build/api.py", line 186, in build
    return build_tree(
  File "/Users/runner/miniforge3/lib/python3.8/site-packages/conda_build/build.py", line 3068, in build_tree
    packages_from_this = build(metadata, stats,
  File "/Users/runner/miniforge3/lib/python3.8/site-packages/conda_build/build.py", line 2191, in build
    utils.check_call_env(cmd, env=env, rewrite_stdout_env=rewrite_env,
  File "/Users/runner/miniforge3/lib/python3.8/site-packages/conda_build/utils.py", line 412, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "/Users/runner/miniforge3/lib/python3.8/site-packages/conda_build/utils.py", line 392, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['/bin/bash', '-o', 'errexit', '/Users/runner/miniforge3/conda-bld/proj_1617263076306/work/conda_build.sh']' returned non-zero exit status 2.
##[error]Bash exited with code '1'.
##[section]Finishing: Run OSX build
```

**Comment by Tobias-Fischer** [comment]
@mwtoews: It's picking up the wrong sqlite executable:
```
2021-04-01T09:17:24.7711960Z /bin/sh: /Users/runner/miniforge3/conda-bld/proj_1617268421789/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_plac/bin/sqlite3: Bad CPU type in executable
```

The cmake uses `find_program(EXE_SQLITE3 sqlite3)`. This can probably avoided by passing `-EXE_SQLITE3=$BUILD_PREFIX/bin/sqlite3` to the cmake call.

**Comment by Tobias-Fischer** [comment]
Looks like this worked :). Now it's a matter of doing sth like
```sh
if [[ "$CONDA_BUILD_CROSS_COMPILATION" != "1" ]]; then
     EXE_SQLITE3=${PREFIX}/bin/sqlite3
else
     EXE_SQLITE3=${BUILD_PREFIX}/bin/sqlite3
fi

cmake ....  -D EXE_SQLITE3=$EXE_SQLITE3 ....
```

**Comment by mwtoews** [comment]
Thanks again @Tobias-Fischer, I wrote similar `if/fi` statement in parallel, but I'll take yours instead. Apologies for all the `push -f`'s ..., it seems a stable draft PR is not far off.

**Comment by Tobias-Fischer** [comment]
Nice! For some reason most builds weren't triggered for the last push. Could you do an empty commit?

Would be cool if this PR could be merged, I don't see any disadvantages.

**Comment by xylar** [comment]
As I commented [here](https://github.com/conda-forge/proj.4-feedstock/issues/105#issuecomment-811697860) and was discussed in relation to GEOS [here](https://github.com/conda-forge/geos-feedstock/pull/60#issuecomment-811503630), we need to be careful that switching to CMake doesn't change the ABI in unexpected ways, and it might be safest to wait for a new release.

On another note, I think it's important to rerender for every PR where the build number gets bumped.

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
I should have said up front, thank you very much for working on this, @mwtoews!

**Comment by mwtoews** [comment]
Given that that we're not 100% certain of ABI compatibility between the builds, I'd be happy to wait to merge this (with an additional commit) for PROJ 8.1.0 planned for 1st of July 2021.

**Comment by mwtoews** [comment]
PROJ 8.1.0 [has been released](https://lists.osgeo.org/pipermail/proj/2021-June/010305.html). How should we proceed? Add a commit to this PR? Create a new PR for the new release with these changes?

**Comment by Tobias-Fischer** [comment]
I'm not a maintainer of this package, but adding a commit to this PR + Changing the description sounds good to me.

**Comment by mwtoews** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
@mwtoews, the bot's update to v8.1.0 got automatically merged (which I don't think we wanted) in #108.  Given that, maybe the easiest thing is to bump the build number to 1 here.  Would you be willing to do that.

I also think we probably want to turn off the automerge bot.  If @ocefpaf agrees, maybe we can add that to this PR, too.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
   name: proj
   version: {{ version }}
 
 source:
   url: http://download.osgeo.org/proj/proj-{{ version }}.tar.gz
-  sha256: e0463a8068898785ca75dd49a261d3d28b07d0a88f3b657e8e0089e16a0375fa
+  sha256: 22c5cdc5aa0832077b16c95ebeec748a0942811c1c3438c33d43c8d2ead59f48
 
 build:
   number: 0
```
Comment (ocefpaf) [inline recipe/meta.yaml]: The automerge PR created a build 0 for this version already. Let's bump here to 1/

**Pull request merged**